app/src/main/java/org/schabi/newpipe/streams/OggFromWebMWriter.java

```diff
@@ -0,0 +1,434 @@
+
+            if (bloq == null) {
+                packet_flag = FLAG_LAST;// note: if the flag is FLAG_CONTINUED, is changed
+                elapsed_ns += webm_block_last_timecode;
+
+                if (webm_track.defaultDuration > 0) {
+                    elapsed_ns += webm_track.defaultDuration;
+                } else {
+                    // hardcoded way, guess the sample duration
+                    elapsed_ns += webm_block_near_duration;
+                }
+            } else {
+                elapsed_ns += bloq.absoluteTimeCodeNs;
+            }
+
+            // get the sample count in the page
+            elapsed_ns = elapsed_ns / TIME_SCALE_NS;
+            elapsed_ns = Math.ceil(elapsed_ns * resolution);
+
+            // create header and calculate page checksum
+            int checksum = make_packetHeader((long) elapsed_ns, header, null);
+            checksum = calc_crc32(checksum, page.array(), page.position());
+
+            header.putInt(HEADER_CHECKSUM_OFFSET, checksum);
+
+            // dump data
+            write(header);
+            write(page);
+
+            webm_block = bloq;
+        }
+    }
+
+    private int make_packetHeader(long gran_pos, @NonNull ByteBuffer buffer, byte[] immediate_page) {
+        short length = HEADER_SIZE;
+
+        buffer.putInt(0x5367674f);// "OggS" binary string in little-endian
+        buffer.put((byte) 0x00);// version
+        buffer.put(packet_flag);// type
+
+        buffer.putLong(gran_pos);// granulate position
+
+        buffer.putInt(STREAM_ID);// bitstream serial number
+        buffer.putInt(sequence_count++);// page sequence number
+
+        buffer.putInt(0x00);// page checksum
+
+        buffer.put((byte) segment_table_size);// segment table
+        buffer.put(segment_table, 0, segment_table_size);// segment size
+
+        length += segment_table_size;
+
+        clearSegmentTable();// clear segment table for next header
+
+        int checksum_crc32 = calc_crc32(0x00, buffer.array(), length);
+
+        if (immediate_page != null) {
+            checksum_crc32 = calc_crc32(checksum_crc32, immediate_page, immediate_page.length);
+            buffer.putInt(HEADER_CHECKSUM_OFFSET, checksum_crc32);
+            segment_table_next_timestamp -= TIME_SCALE_NS;
+        }
+
+        return checksum_crc32;
+    }
+
+    @Nullable
+    private byte[] make_metadata() {
+        if ("A_OPUS".equals(webm_track.codecId)) {
+            return new byte[]{
+                    0x4F, 0x70, 0x75, 0x73, 0x54, 0x61, 0x67, 0x73,// "OpusTags" binary string
+                    0x07, 0x00, 0x00, 0x00,// writting application string size
+                    0x4E, 0x65, 0x77, 0x50, 0x69, 0x70, 0x65,// "NewPipe" binary string
+                    0x00, 0x00, 0x00, 0x00// additional tags count (zero means no tags)
+            };
+        } else if ("A_VORBIS".equals(webm_track.codecId)) {
+            return new byte[]{
+                    0x03,// ????????

```

Comment:
?

Reply:
probably is the vorbis header version or the firs tag count. I do not know